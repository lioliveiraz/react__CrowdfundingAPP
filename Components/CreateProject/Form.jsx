import React, { useState } from 'react';
import Input from '../Input/Input';
import { postProject } from '../../pages/api/fetchData/requests';
import { validateUserInput } from '../../services/validateUserInput';
import { toast, ToastContainer } from 'react-toastify';

function Form(props) {
    const [userInputs, setUserInputs] = useState({
        name: "",
        goal: 0,
        description: "",
        creator: "",
        image: "placeholder.jpg",
        donated: 0

    });
    const [error, setError] = useState({});

    const handleUserInput = (value, name) => {
        setUserInputs({ ...userInputs, [name]: value });
    };

    const handleSubmit = async (e) => {
        e.preventDefault();
        const inputErrors = validateUserInput(userInputs);
        if (inputErrors) {
            setError(inputErrors);
            toast.error(`Ops! Something is not right`);

        } else {
            userInputs["goal"] = parseFloat(userInputs['goal']);
            const response = await postProject(userInputs);
            toast.success(`Thank you, ${response.creator}.The project ${response.name} was created`, { position: "top-right" });


        }

    };

    return (
        <>
            <form data-testid="form" onSubmit={handleSubmit}>
                <Input type="text" label="name" handleChange={handleUserInput} name="name" />
                {error.name && <p style={{ color: "red", fontWeight: "bold" }}>{error.name}</p>}
                <Input type="text" label="creator" name="creator" handleChange={handleUserInput} />
                {error.creator && <p style={{ color: "red", fontWeight: "bold" }}>{error.creator}</p>}
                <label htmlFor="description"> Description</label>
                <textarea id="description" name="description" onChange={(e) => handleUserInput(e.target.value, e.target.name)} />
                {error.description && <p style={{ color: "red", fontWeight: "bold" }}>{error.description}</p>}
                <Input type="number" handleChange={handleUserInput} label="goal" name="goal" />
                {error.goal && <p style={{ color: "red", fontWeight: "bold" }}>{error.goal}</p>}
                <input type="submit" />

            </form>
            <ToastContainer />

        </>

    );
}

export default Form;