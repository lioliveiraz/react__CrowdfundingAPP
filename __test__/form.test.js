import Form from './../Components/CreateProject/Form';
import { mount, shallow } from 'enzyme';
import { act } from "@testing-library/react";

describe('<Form>', () => {
    let target, wrapper;
    target = {
        name: "string",
        creator: "string",
        goal: "string",
        description: "string"

    };

    it('should add items', () => {

        wrapper = shallow(
            <Form />
        );

        expect(wrapper.getElement()).toEqual();


    });
});