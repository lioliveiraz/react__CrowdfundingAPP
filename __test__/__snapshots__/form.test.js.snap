// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form> should add items 1`] = `ShallowWrapper {}`;
